source/code/troubleshooter/omsagent_tsg_zip/tsg_files/tsg_code/tsg_errors.py

```diff
@@ -0,0 +1,323 @@
+          "\n  $ sudo yum downgrade curl libcurl\n\n"\
+          "After either option, please restart OMI using this command:\n"\
+          "\n  $ sudo scxadmin -restart\n\n"\
+          "You can read more about how to fix this specific bug by going to:\n"\
+          "\n    https://github.com/microsoft/OMS-Agent-for-Linux/blob/master/docs/"\
+                    "Troubleshooting.md#i-see-omiagent-using-100-cpu",
+    144 : "There seems to be an issue similar to a common issue involving OMI agent using "\
+          "100% CPU. Please check the below link for more information:\n"\
+          "\n    https://github.com/microsoft/OMS-Agent-for-Linux/blob/master/docs/"\
+                    "Troubleshooting.md#i-see-omiagent-using-100-cpu",
+    145 : "Ran into the following error when trying to run slabtop: \n  {0}",
+    146 : "Your machine has an issue with the dentry cache becoming bloated. Please check the "\
+          "top 10 caches below, sorted by cache size:\n{0}",
+    147 : "Your version of nss-softokn is slightly out of date, causing an issue with "\
+          "bloating the dentry cache. Please upgrade your version to nss-softokn-3.14.3-12.el6 "\
+          "or newer, and ensure that the NSS_SDB_USE_CACHE environment variable is set to 'yes'. "\
+          "Please check the below link for more information:\n"\
+          "\n    https://bugzilla.redhat.com/show_bug.cgi?format=multiple&id=1044666",
+    148 : "There appears to be an issue in NSS, which resulted in bloating the dentry cache. "\
+          "Please set the NSS_SDB_USE_CACHE environment variable to 'yes'. You can check the "\
+          "below link for more information:\n"\
+          "\n    https://github.com/microsoft/OMS-Agent-for-Linux/blob/master/docs/"\
+                    "Troubleshooting.md#i-see-omiagent-using-100-cpu",
+    149 : "Logrotate size limit for log {0} has invalid formatting. Please see {1} for more "\
+          "information.",
+    150 : "Logrotate isn't rotating log {0}: its current size is {1}, and it should have "\
+          "been rotated at {2}. Please see {3} for more information.",
+    151 : "File {0} has been modified {1} times in the last {2} seconds.",
+    152 : "{0} isn't installed correctly.",
+    153 : "The agent is currently taking too long to pull the configuration from the backend. "\
+          "Please check the following log files for more information:\n"\
+          "    {0}\n"\
+          "    {1}\n"\
+          "Please also check out the below link (step 6) for more information:\n"\
+          "\n    https://supportability.visualstudio.com/AzureLogAnalytics/_wiki/wikis/"\
+                    "Azure-Log-Analytics.wiki/168967/Custom-Logs-Linux-Agent",
+    154 : "This version of Python is missing the {0} package. (You can check by opening up "\
+          "python and typing 'import {0}'.) Please install this package and run the troubleshooter again."
+
+}  # TODO: keep up to date with error codes onenote
```

Comment:
Might be better to maintain an .md in this repo publicly so that all users can see?

Reply:
Added it to the TST Github documentation here:
https://github.com/microsoft/OMS-Agent-for-Linux/blob/bff7c80f869d6f5a4046ed41b4bd0ed89d69219f/docs/Troubleshooting-Tool.md#list-of-possible-errors